### ISSUE TITLE
BMM150 still not working after i2c patch from sumit

### ISSUE THREAD

**Issue by landonh12** [NXPHoverGames/zephyr#6]
BME280 works, but BMM150 does not.

By changing the BMM150 sample to follow the same device tree api as the BME280, we can observe that the bmm150 gets recognized, but we get a divide by zero error:

```
[0*** Booting Zephyr OS build zephyr-v3.0.0-3207-gf7f77f259711  ***
BMM150 Geomagnetic sensor Application
Found device is 0xa0c4, name is bmm150
0:00:00.000,[00:00:00.000,000] <inf> regulator_fixed: reg-3v3-neth onoff: 0
[00:00:00.000,000] <inf> regulator_fixed: reg-3v3-s onoff: 0
[00:00:00.000,000] <err> BMM150: failed to bring up device from normal mode
[00:00:00.000,000] <err> BMM150: failed to initialize chip
[00:00:00.014,000] <err> os: ***** USAGE FAULT *****
[00:00:00.014,000] <err> os:   Division by zero
[00:00:00.014,000] <err> os: r0/a1:  0x20000978  r1/a2:  0x00000000  r2/a3:  0x00000000
[00:00:00.014,000] <err> os: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x00000000
[00:00:00.014,000] <err> os:  xpsr:  0x41000000
[00:00:00.014,000] <err> os: Faulting instruction address (r15/pc): 0x00008acc
[00:00:00.014,000] <err> os: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:00:00.014,000] <err> os: Current thread: 0x200006e8 (unknown)
[00:00:00.096,000] <err> os: Halting system
```

After stepping through code, we can find that the sample is trying to access BMM150 register data before the registers are read for the first time. Therefore, all registers are set to 0, and we run into a divide by zero error on line 251 of bmm150.c

```
	prevalue = (uint16_t)((((int32_t)tregs->xyz1) << 14) / rhall);
```

It seems apparent that there is still some priority issues between the I2C driver, BMM150, and GPIO (regulator-fixed).

**Comment by landonh12** [comment]
So I found this file from another board: 

https://github.com/NXPHoverGames/zephyr/blob/rddrone_fmuk66-i2c/boards/arm/thingy53_nrf5340/board.c

In it, it states:

/* Initialization chain of Thingy:53 board requires some delays before on board sensors
 * could be accessed after power up. In particular bme680 and bmm150 sensors require,
 * respectively 2ms and 1ms power on delay. In order to avoid delays sum, common delay is
 * introduced in the board start up file. Below asserts ensure correct initialization order:
 * on board regulators, board init (this), sensors init.
 */

Bingo! We just need to create a similar configuration for rddrone_fmuk66.

**Comment by sumitbatra-nxp** [comment]
> So I found this file from another board:
> 
> https://github.com/NXPHoverGames/zephyr/blob/rddrone_fmuk66-i2c/boards/arm/thingy53_nrf5340/board.c
> 
> In it, it states:
> 
> ```
> /* Initialization chain of Thingy:53 board requires some delays before on board sensors
>  * could be accessed after power up. In particular bme680 and bmm150 sensors require,
>  * respectively 2ms and 1ms power on delay. In order to avoid delays sum, common delay is
>  * introduced in the board start up file. Below asserts ensure correct initialization order:
>  * on board regulators, board init (this), sensors init.
>  */
> ```
> 
> Bingo! We just need to create a similar configuration for rddrone_fmuk66. @sumitbatra-nxp, I think this is the correct way to do this, rather than editing the gpio_mcux file.

thanks a ton @landonh12 
this is an awesome find..  with these changes both bmm150 and bme280 work fine

**Comment by sumitbatra-nxp** [comment]
I think these should go in the form of 2 different patches - 
1) Fix for rddrone_fmuk66 I2C sensors enablement
2) bmm150 sample driver detect issue

**Comment by sumitbatra-nxp** [comment]
commit ed7841a3fc169107e64450a0b681b6d041e91399

**Comment by sumitbatra-nxp** [comment]
> 
Good point..  I agree with your changes..
Let me pen down what I understood from this Sensor Enablement Exercise -
1) Amongst board, regulator and bmm/bme sensor modules - Regulator module gets initialized first which enables the 3.3V for the sensors. 
2) Then board module gets into picture and introduces a delay of 2ms before respective sensor module initialization.
3) Finally sensor (bmm/bme and others) modules get initialized (device and driver binding also happens) which involves I2C communication as well.

**Comment by sumitbatra-nxp** [comment]
So will you generate a pull request ? Or shall I do it ?

**Comment by landonh12** [comment]
I think if we make a pull request, we should make sure that we don't do it until we have all the interfaces working. That way we don't have to do multiple pull requests for the board.

Nothing is urgent right now as nobody is using Zephyr for FMUK66 yet.

**Comment by landonh12** [comment]
Squashed commits are merged to main in 0e59cac2f9f86e8bdc88aa690fd0dc38ed78220f

https://github.com/NXPHoverGames/zephyr/pull/7

**Comment by landonh12** [comment]
Squashed commits are merged to main in 0e59cac2f9f86e8bdc88aa690fd0dc38ed78220f

https://github.com/NXPHoverGames/zephyr/pull/7
